osi_logicallane.proto

```diff
@@ -0,0 +1,502 @@
+    //   distance == euclidean distance between the points taking only X and Y
+    //   into account).
+    // - The S distance between two points may be larger than the 2D euclidean
+    //   distance, but should be not much larger. It is allowed to be larger if
+    //   the underlying reference line (e.g. in an OpenDRIVE map) is a curve,
+    //   and thus the sampled road has a smaller length than the original
+    //   curve.
+    //
+    // Together, these rules allow directly putting OpenDRIVE S coordinates
+    // into an OSI ReferenceLine.
+    //
+    // If the reference line approximates a curve (e.g. a clothoid in
+    // OpenDRIVE), the points must be chosen in a way that the lateral distance
+    // to the ideal line does not exceed 5cm. As shown in the following image:
+    //
+    // \image html line_approximation_error.svg "Approximation error"
+    // Approximation error green line.
+    //
+    // Between two ReferenceLinePoints, both the world coordinate and the S
+    // coordinate is interpolated linearly. So each S value uniquely describes
+    // a point on the polyline.
+    //
+    // For the purpose of this discussion, let's call the S position of the
+    // first point sStart, and the S position of the last point sEnd.
+    //
+    // For some purposes, S positions outside the normally defined range (i.e.
+    // outside [sStart,sEnd]) need to be defined. For this purpose, the first
+    // line of the polyline is infinitely extended in negative S direction.
+    // Similarly, the last line of the polyline is infinitely extended beyond
+    // the last point. The S value of points outside [sStart,sEnd] is defined
+    // by the euclidean 2D distance from the start or end point, respectively.
+    // So if sStart = 15, and a point is on the line extended from the start
+    // position, with a 2D euclidean distance of 10 from the first point, then
+    // it has an S position of 5.
+    //
+    // A point is "before" the reference line, if its s coordinate is < sStart.
+    // A point is "after" the reference line, if its s coordinate is > sEnd.
+    //
+    // To describe points that are not directly on the polyline, a T
+    // coordinate is added. T is the signed 2D distance (i.e. hypot(A.X-B.X,
+    // A.Y-B.Y), if A and B are the two points) between the point to describe
+    // and the nearest point on the polyline (this point might either be on a
+    // line segment or at an edge between two line segments). The distance is
+    // positive if the point is left of the polyline (in definition direction),
+    // negative if it is right of it. Note that the "nearest point on the
+    // polyline" is calculated in 3D, in order to choose the correct point for
+    // 3D curves (think reference lines for roads in parking decks). If there
+    // are several "nearest points", the one with the smallest S coordinate on
+    // the polyline is chosen.
```

Comment:
TODO: At this point I could imagine to incorporate Daniel's images on that topic.

Reply:
I added a similar image, roughly based on the ones from @dbeckerAC . I think the text from Daniel could be added as an additional separate text, but I won't do this as part of this change (because of lack of time). I think this can be done later on-top.